marker_rustc_driver/src/main.rs

```diff
@@ -15,6 +15,14 @@
 #![allow(clippy::needless_lifetimes, reason = "lifetimes will be required to fix ICEs")]
 #![allow(clippy::needless_collect, reason = "false positives for `alloc_slice`")]
 #![allow(clippy::too_many_lines, reason = "long functions are unavoidable for matches")]
+// Custom Marker lints
+#![cfg_attr(
+    marker,
+    feature(register_tool),
+    register_tool(marker),
+    // This lint applies only to public `marker_api`
+    allow(marker::not_using_has_span_trait)
```

Comment:
It might be better, to have the lint allow-by-default and then add the `warn(marker::not_using_has_span_trait)` to `marker_api` instead. I also thought about adding a lint crate for internal marker lints. :thinking:

Reply:
Good idea 💡, I forgot about that